test/libsolidity/SolidityEndToEndTest.cpp

```diff
@@ -7173,6 +7173,24 @@ BOOST_AUTO_TEST_CASE(cross_contract_types)
 	ABI_CHECK(callContractFunction("f()"), encodeArgs(u256(3)));
 }
 
+BOOST_AUTO_TEST_CASE(cross_contract_constants)
+{
+	char const* sourceCode = R"(
+		contract C {
+			uint constant a = 33 * b + 4;
+			uint constant b = 44;
+		}
+		contract Test {
+			uint constant b = 66;
+			function f() returns (uint r) {
+				r = C.a;
+			}
+		}
+	)";
+	compileAndRun(sourceCode, 0, "Test");
+	ABI_CHECK(callContractFunction("f()"), encodeArgs(u256(1456)));
+}
+
```

Comment:
Are there any other weird cases we have to test? Is it somehow possible to use `super` or `this` in that context?

Reply:
Is it possible to access constants via `super`? If yes, does that work correctly if that constant is accessed from another contract?

Reply:
Access via ``super`` is not allowed so far - I think it'd be better to have to use the name of the base explicitly instead. I can change that, though, if you think otherwise. In any case I added a test (for now with the expectation that access through super is not possible).